crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -1809,6 +1813,20 @@ impl<TX: DbTxMut + DbTx + 'static, N: NodeTypesForProvider> StateWriter
             .then(|| self.static_file_provider.get_writer(first_block, StaticFileSegment::Receipts))
             .transpose()?;
 
+        let has_contract_log_filter = !self.prune_modes.receipts_log_filter.is_empty();
```

Comment:
doing this here makes so much more sense because we already have the prune_modes here